src/measure_theory/integral/integrable_on.lean

```diff
@@ -246,6 +246,37 @@ begin
   exact ((Lp.mem_ℒp _).restrict s).mem_ℒp_of_exponent_le hp,
 end
 
+lemma integrable_on.lintegral_lt_top_ae' {f : α → ℝ} {s : set α}
+  (hf : integrable_on f s μ) (h_nonneg : 0 ≤ᵐ[μ.restrict s] f) :
+  ∫⁻ x in s, ennreal.of_real (f x) ∂μ < ⊤ :=
```

Comment:
You don't even need the hypothesis `h_nonneg`:
```lean
variables {α : Type*} {m : measurable_space α} {μ : measure α}

lemma lintegral_to_real_le_lintegral_to_real_norm (f : α → ℝ) :
  ∫⁻ x, ennreal.of_real (f x) ∂μ ≤ ∫⁻ x, ennreal.of_real (∥f x∥) ∂μ :=
begin
  refine lintegral_mono (λ x, ennreal.of_real_le_of_real _),
  rw real.norm_eq_abs,
  exact le_abs_self (f x),
end

lemma integrable_on.lintegral_lt_top_ae' {f : α → ℝ} {s : set α} (hf : integrable_on f s μ) :
  ∫⁻ x in s, ennreal.of_real (f x) ∂μ < ∞ :=
calc ∫⁻ x in s, ennreal.of_real (f x) ∂μ
    ≤ ∫⁻ x in s, ennreal.of_real (∥f x∥) ∂μ : lintegral_to_real_le_lintegral_to_real_norm f
... = ∫⁻ x in s, ↑∥f x∥₊ ∂μ : by simp_rw ← of_real_norm_eq_coe_nnnorm
... < ∞ : hf.2
```
Also please use `∞` instead of `⊤` for ennreal.

Reply:
Yes, it makes totally sense that very negative values are bad for the integral.

Note that [has_finite_integral](https://leanprover-community.github.io/mathlib_docs/measure_theory/function/l1_space.html#measure_theory.has_finite_integral) uses consistently `top`, so it might be good to change it there, too.

Reply:
`has_finite_integral` is one of the oldest objects we have about integration. The convention was maybe not in place at that time. I'll PR a change of notation there.